File: src/Web/JavaScriptJS/singleton.js
class ProcessRegistry {
  constructor() {
    if (ProcessRegistry.instance) return ProcessRegistry.instance;
    this.count = 0;
    ProcessRegistry.instance = this;
  }

  increment() {
    this.count += 1;
  }
}

const first = new ProcessRegistry();
const second = new ProcessRegistry();
first.increment();
console.log(`same=${first === second}`);
console.log(`count=${second.count}`);

File: src/Scripting/PythonPY/singleton.py
class ProcessRegistry:
    _instance = None

    def __new__(cls):
        if cls._instance is None:
            cls._instance = super().__new__(cls)
            cls._instance.count = 0
        return cls._instance

    def increment(self):
        self.count += 1


first = ProcessRegistry()
second = ProcessRegistry()
first.increment()
print(f"same={str(first is second).lower()}")
print(f"count={second.count}")
